packages/core/src/components/DataTable/helpers/sortData.ts

```diff
@@ -33,3 +33,29 @@ export default function sortData<T extends GenericRow>(
 
   return list;
 }
+
+export default function sortData<T extends GenericRow>(
+  list: T[],
+  keys: string[],
+  selectedRows: SelectedRows,
+  selectedRowsFirst: boolean,
+  sortBy?: string,
+  sortDirection?: SortDirectionType,
+): T[] {
+  if (selectedRowsFirst) {
+    const selectedList = list.filter((row: T) =>
```

Comment:
Instead of doing 2 filter loops, I would suggest just doing a `forEach` and pushing the item to either the selected or non-selected lists.

Reply:
Good point, got it!